api/applicationserver.proto

```diff
@@ -31,14 +31,8 @@ package ttn.lorawan.v3;
 option go_package = "go.thethings.network/lorawan-stack/v3/pkg/ttnpb";
 
 message ApplicationLink {
-  // The address of the external Network Server where to link to.
-  // The typical format of the address is "host:port". If the port is omitted,
-  // the normal port inference (with DNS lookup, otherwise defaults) is used.
-  // Leave empty when linking to a cluster Network Server.
-  string network_server_address = 1 [(validate.rules).string.pattern = "^(?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*(?:[A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])(?::[0-9]{1,5})?$|^$"];
-  // The API key to use to link the Application Server to Network Server.
-  // This API key needs to have RIGHT_APPLICATION_LINK.
-  string api_key = 2 [(gogoproto.customname) = "APIKey", (validate.rules).string.min_len = 1];
+  reserved 1; // Deprecated: network_server_address
+  reserved 2; // Deprecated: api_key
```

Comment:
I'm pretty sure this violates our compatibility commitment, as existing scripts that rely on the CLI or API will now get validation errors when getting the link (because these fields are no longer allowed to be in the field mask). Similarly, when setting the link I think the server should ignore the fields instead of erroring (possibly returning a warning in the header/trailer).

Reply:
So should we keep them here?

Reply:
The last commit leaves these fields as part of the allowed field mask and removes them before calling the store, while generating a warning about the field's deprecated status.